packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -101,7 +107,7 @@ export const assertIssuerKeywords = (zcf, expected) => {
  * (update's values prevailing if the keywords are the same)
  * to produce the newAllocation.
  * @param {ContractFacet} zcf
- * @param {ZCFSeat} seat
+ * @param {ZcfSeatPartial} seat
```

Comment:
Hmm, why do we need this new type? This should just be a real ZCFSeat.

Reply:
In unit tests, particularly with dependency injection, it's often useful to declare an interface type that only has the subset of methods that are actually relied on by a function. 

In the test 'ZoeHelpers satisfies simple proposal', I create fake seats that have all the behavior this test relies on in two lines. If the test had to build actual ZcfSeats it would be much more involved, and it would be harder to build arbitrary combinations with arbitrary behavior.

Reply:
> In unit tests, particularly with dependency injection, it's often useful to declare an interface type that only has the subset of methods that are actually relied on by a function.

Yup, that makes sense. No arguments there.

I do wonder about our Typescript/JSDoc checking though. In the past, I observed that an object that had properties not listed in the type was *not* of that type. Is it true that we can pass in a full ZCFSeat here without penalty even though the type is a subset of the properties of a ZCFSeat? I would be surprised, but perhaps that just means that I don't understand what the criteria for types are. @michaelfig could you confirm? Thanks!

Reply:
Just my $0.02 until @michaelfig confirms.

Partial<Foo> is less-that-or-equal like rather than less-than like. It means it'll accept any subset of those properties, including all of them.

I do not like the idea of weakening the typing of production code in order to accommodate mocks. The types are about what correct code production code should be. Mocks are supposed to take shortcuts. I would prefer putting types in production code for production code, and then using `// eslint-disable-next-line` as appropriate in the mock.

Reply:
> I do not like the idea of weakening the typing of production code in order to accommodate mocks. The types are about what correct code production code should be. Mocks are supposed to take shortcuts. I would prefer putting types in production code for production code, and then using // eslint-disable-next-line as appropriate in the mock.

I agree. It seems like it would really nice to have some additional types that can be used in testing to test our assumptions of what methods are being used, but I think we should not change the user-facing types to do so.

Reply:
The fact that we even have the `eslint-disable` option is a testament to what TypeScript got right that almost all statically typed languages get wrong. Code can be run that doesn't pass the static check. It can even say that it does so on purpose, so stop warning me.

Reply:
> `Partial<Foo>` is less-that-or-equal like rather than less-than like. It means it'll accept any subset of those properties, including all of them.

Confirmed.

> I do not like the idea of weakening the typing of production code in order to accommodate mocks. The types are about what correct code production code should be. Mocks are supposed to take shortcuts. I would prefer putting types in production code for production code, and then using `// eslint-disable-next-line` as appropriate in the mock.

I think you mean `// @ts-ignore`.  ESLint is separate from Typescript.